client/js/users.js

```diff
@@ -0,0 +1,161 @@
+            listPost2.innerText = `Age: ${users.age}`;
+            searchDiv.append(listPost, listPost2)
+        })
+    }
+}
+
+
+async function searchTopLeftUser() {
+    event.preventDefault();
+    let userName = document.querySelector('#search2').value;
+    const findAllUsers = await axios.get(`http://localhost:11000/users/`)
+    let responsedata = findAllUsers.data.body
+    let userId = searchNames(responsedata, userName)
+    const title = document.getElementById("allTheUsers")
+    title.innerHTML = `Results for ${userName}`;
+    const userPost = document.querySelector("#usersResults");
+    userPost.innerHTML = ""
+    for (let i = 0; i < userId.length; i++) {
+        let response = await axios.get(`http://localhost:11000/users/${userId[i]}`)
+        response.data.body.forEach((users) => {
+            let searchDiv = document.createElement('div')
+            searchDiv.className = "searchResultsDiv"
+            userPost.append(searchDiv)
+            let listPost = document.createElement('li')
+            let listPost2 = document.createElement('li')
+            listPost.innerText = `Name:${users.firstname} ${users.lastname}`;
+            listPost2.innerText = `Age: ${users.age}`;
+            searchDiv.append(listPost, listPost2)
+        })
+    }
+}
+
+const searchNames = (data, userName) => {
+    let id = []
+    for (result of data) {
+        if (userName.toUpperCase() === result.firstname.toUpperCase()) {
+            id.push(result.user_id)
+        } else if (userName.toUpperCase() === result.lastname.toUpperCase()) {
+            id.push(result.user_id)
+        } else if (userName.toUpperCase() === result.firstname.toUpperCase() + " " + result.lastname.toUpperCase()) {
+            id.push(result.user_id)
+        }
+    }
+    return id
+}
+
+const addAUser = async () => {
+    event.preventDefault();
+    const firstname = document.querySelector("#searchBoxAddFN").value
+    const lastname = document.querySelector("#searchBoxAddLN").value
+    const age = document.querySelector("#searchBoxAddAge").value
+    const findAllUsers = await axios.get(`http://localhost:11000/users/`)
+    let responseData = findAllUsers.data.body
+    let id = filerResults(responseData, firstname, lastname, age)
+    if (id === -1) {
+        await axios.post(`http://localhost:11000/users/`, { firstname, lastname, age });
+        loadUsers()
+    }
+}
+
+
+const deleteAUser = async () => {
+    event.preventDefault();
+    let firstname = document.querySelector('#searchBoxDeleteFN').value;
+    let lastname = document.querySelector('#searchBoxDeleteLN').value;
+    let age = document.querySelector('#searchBoxDeleteAge').value;
+    const findAllUsers = await axios.get(`http://localhost:11000/users/`)
+    let responseData = findAllUsers.data.body
+    let index = filerResults(responseData, firstname, lastname, age)
+    let user_id = responseData[index].user_id
```

Comment:
if the queried user entry is not found, your code breaks here because it will say 'cannot read property user.id of undefined'. (Like try entering the wrong age). To make your code work the way it is i would shift line 134 down into the if conditional
```suggestion
    if (index !== -1) {
        let user_id = responseData[index].user_id
```

however, the userSimulator code I made would let you grab the current user's id number and you could use that to find their account instead of checking firstname, lastname, age. paste the following line into your head to use the script on your page and then you could querySelector('#userNum') for their user id
`   <script src="../js/userSim.js" type="text/javascript"></script>`